### PR TITLE
Fix data races in tests, closes #255

### LINKED ISSUE #255: Data race detected in TestLatestTagSearchHTTP test
Reproducible 100% of the time on 10.195.70.248 (48 cores bare metal).
For login details contact me on the chat

Full logs can be found here : https://gist.github.com/adodon2go/d7976bad96b9f3814fd9d77f76680e4c

### PR DESCRIPTION
**Pull request by adodon2go**
Signed-off-by: Alexei Dodon <adodon@cisco.com>

Root cause: same port used by controller & shared across different tests that runs in parallel.

### CODE REVIEW

File: pkg/extensions/search/common/common_test.go [hunk 1]
```diff
@@ -28,8 +30,7 @@ var (
 )
 
 const (
-	BaseURL1 = "http://127.0.0.1:8085"
-	Port1    = "8085"
+	BaseURL = "http://127.0.0.1:%s"
```
Comment (rchincha) [inline pkg/extensions/search/common/common_test.go:33]: Unless you are referring to this directly, just move this into getBaseURL() itself.

Comment (rchincha) [reply pkg/extensions/search/common/common_test.go:33]: Same comment in other test files as well.

File: pkg/extensions/search/common/common_test.go [hunk 2]
```diff
@@ -57,6 +58,19 @@ type ImageInfo struct {
 	Labels      string
 }
 
+func getFreePort() string {
```
Comment (rchincha) [inline pkg/extensions/search/common/common_test.go:61]: Thanks for this cleanup.

File: pkg/extensions/search/digest/digest_test.go [hunk 3]
```diff
@@ -29,8 +31,7 @@ var (
 )
 
 const (
-	BaseURL1 = "http://127.0.0.1:8085"
-	Port1    = "8085"
+	BaseURL = "http://127.0.0.1:%s"
```
Comment (rchincha) [inline pkg/extensions/search/digest/digest_test.go:34]: Move this into getBaseURL() also?

File: pkg/extensions/search/common/common_test.go (hunk 2 above)
Comment (rchincha) [reply pkg/extensions/search/common/common_test.go:61]: Wondering this needs to move into some internal/ or pkg/test/ so code doesn't get repeated.

**Review by peusebiu — APPROVED** [review]

**Review by shimish2 — APPROVED** [review]
lgtm

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**